addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationHandlerFactory.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.internal.kostal.inverter.thirdgeneration;
+
+import static org.openhab.binding.internal.kostal.inverter.thirdgeneration.ThirdGenerationBindingConstants.*;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link ThirdGenerationHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author René - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.kostalpikoiqplenticore", service = ThingHandlerFactory.class)
+public class ThirdGenerationHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = new HashSet<ThingTypeUID>(
+            Arrays.asList(PIKOIQ42, PIKOIQ55, PIKOIQ70, PIKOIQ85, PIKOIQ100, PLENTICOREPLUS42WITHBATTERY,
+                    PLENTICOREPLUS55WITHBATTERY, PLENTICOREPLUS70WITHBATTERY, PLENTICOREPLUS85WITHBATTERY,
+                    PLENTICOREPLUS100WITHBATTERY, PLENTICOREPLUS42WITHOUTBATTERY, PLENTICOREPLUS55WITHOUTBATTERY,
+                    PLENTICOREPLUS70WITHOUTBATTERY, PLENTICOREPLUS85WITHOUTBATTERY, PLENTICOREPLUS100WITHOUTBATTERY));
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
```

Comment:
How about creating a Map of ThingTypeUID -> ThirdGenerationInverterTypes. This could avoid the ifs

Reply:
Done